pettingzoo/classic/hanabi/hanabi.py

```diff
@@ -196,8 +197,14 @@ def __float__(self):
 
 
 def env(**kwargs):
-    env = r_env = raw_env(**kwargs)
-    env = wrappers.CaptureStdoutWrapper(env)
+    render_mode = kwargs.get("render_mode")
+    if render_mode == "ansi":
+        kwargs["render_mode"] = "human"
+        env = r_env = raw_env(**kwargs)
+        env = wrappers.CaptureStdoutWrapper(env)
+    else:
+        env = r_env = raw_env(**kwargs)
+
     env = wrappers.TerminateIllegalWrapper(
         env, illegal_reward=HanabiScorePenalty(r_env)
```

Comment:
This is not your doing, but the fact that `r_env` is needed here kinda scares me. Does anyone even know what this is doing? Don't `env` and `r_env` just point to the same object?

Reply:
@jjshoots Nope, they don't.

![image](https://user-images.githubusercontent.com/14932240/192850050-4bc789d2-b253-465a-93a6-f4158ffcaaa7.png)

Reply:
I have no idea why there are two variable there, but yes, they point to the same object (@WillDudley you should try with an object)

Reply:
Yeah with objects they usually do. Python is black magic.

Reply:
That said, could we try removing the need for r_env? I've looked through the code several times and can't see any reason it would break without it... but Python is black magic so ¯\_(ツ)_/¯

Reply:
Oh that's interesting and not at all confusing (: 

I assumed (because I thought `env` and `r_env` pointed to different objects), that `r_env` was so that they could do `HanabiScorePenalty(r_env)`, where `r_env` isn't wrapped by `CaptureStdoutWrapper()`. Genuinely doesn't seem to be any other reason for it.

The git blame points to Ben Black so we could ask him, or we could just remove `r_env` entirely (indifferent to which).

Reply:
I removed it; tests are still ok and I am pretty confident saying now it is equivalent to before

Reply:
@WillDudley They do point to the same object. Integers are immutable, so += is not an in-place operation for integers.

```
>>> a = b = []
>>> a.append(1)
>>> b
[1]
>>>
```

Reply:
The reason for this weird double assignment is that we often found ourselves needing to refer to the original raw environment `r_env` for various environment specific attributes when creating some of the wrappers. And the `env` variable would get clobbered. This is before we had the `unwrapped` attribute.

This pattern got copied here, not sure if it was used in this way.